custom/icds_reports/sqldata/exports/growth_tracker_report.py

```diff
@@ -49,12 +54,16 @@ def _fetch_data(filters, order_by, case_by_grouping=False):
         elif self.loc_level == 2:
             filters['district_id'] = location
             order_by = ('block_name', 'supervisor_name', 'awc_name', 'person_name')
-
-        data_month_3 = _fetch_data(filters, order_by)
+        query_filters = [{"filters": filters, "order_by": order_by, "case_by_grouping": False}]
+        filters = filters.copy()
         filters["month"] = initial_month - relativedelta(months=1)
-        data_month_2 = _fetch_data(filters, order_by, True)
+        query_filters.append({"filters": filters, "order_by": order_by, "case_by_grouping": True})
+        filters = filters.copy()
         filters["month"] = initial_month - relativedelta(months=2)
-        data_month_1 = _fetch_data(filters, order_by, True)
+        query_filters.append({"filters": filters, "order_by": order_by, "case_by_grouping": True})
+        jobs = [gevent.spawn(_fetch_data, query_filter) for query_filter in query_filters]
+        gevent.joinall(jobs, timeout=120)
```

Comment:
I don't know believe our celery workers run with gevent patched so I don't know if this will actually do anything. Only certain processes patch gevent and its based on a command whitelist in `manage.py`

Reply:
back to the `gevent` thing for a minute, what happens after 120 seconds with this command? What occurs if the timeout is hit? Does it just continue in the function, even though the queries have not returned? That does not seem safe at all.

Reply:
Yeah, that's part I still wanted to discuss before going forward.